lib/openstack_query/api/query_api.py

```diff
@@ -199,4 +206,79 @@ def to_html(
         :param groups: a list group to limit output by
         :param kwargs: kwargs to pass to generate table
         """
-        return self.output.to_html(self._query_results, title, groups, **kwargs)
+        _, selected_results = self.executer.parse_results()
+        return self.output.to_html(selected_results, title, groups, **kwargs)
+
+    def then(self, query_type: Union[str, QueryTypes], forward_outputs: bool = True):
+        """
+        Public method to chain current query into another query of a different type
+        and return the new query so that it will work only on the results of the original query.
+        NOTE - query must be run first for this to work
+        NOTE - a shared common property must exist between this query and the new query
+            - i.e. both ServerQuery and UserQuery share the 'USER_ID' property so chaining is possible
+                - see Mappings for more chaining options
+
+        :param query_type: an enum representing the new query to chain into
+        :param forward_outputs: (default=True) boolean that if true - will forward outputs from t
+        his query (and previous chained queries) onto new query.
```

Comment:
We should add something about `forward_outputs == False` simply runs another query based on the previous results as a filter without adding the fields to make it clear